core/src/main/java/org/elasticsearch/http/netty/cors/CorsHandler.java

```diff
@@ -131,6 +131,17 @@ private static void forbidden(final ChannelHandlerContext ctx, final HttpRequest
             .addListener(ChannelFutureListener.CLOSE);
     }
 
+    private static boolean isSameOrigin(final String origin, final String host) {
+        if (Strings.isNullOrEmpty(host) == false) {
+            // strip protocol from origin
+            final String originDomain = origin.replaceFirst("(http|https)://", "");
```

Comment:
My first thought was that the regex `^https?://` is probably faster. A simple benchmark showed it to be around 40% faster. My second thought was it's shame that we have to create so much garbage here (compiling a regex, and creating the substring). But we do not need to compile the regex every time, we can just create a static regex:

```diff
diff --git a/core/src/main/java/org/elasticsearch/http/netty/cors/CorsHandler.java b/core/src/main/java/org/elasticsearch/http/netty/cors/CorsHandler.java
index 4eb6323..2c8a575 100644
--- a/core/src/main/java/org/elasticsearch/http/netty/cors/CorsHandler.java
+++ b/core/src/main/java/org/elasticsearch/http/netty/cors/CorsHandler.java
@@ -33,6 +33,7 @@ import org.jboss.netty.handler.codec.http.HttpResponse;
 
 import java.util.HashSet;
 import java.util.Set;
+import java.util.regex.Pattern;
 
 import static org.jboss.netty.handler.codec.http.HttpHeaders.Names.ACCESS_CONTROL_ALLOW_CREDENTIALS;
 import static org.jboss.netty.handler.codec.http.HttpHeaders.Names.ACCESS_CONTROL_ALLOW_HEADERS;
@@ -131,10 +132,12 @@ public class CorsHandler extends SimpleChannelUpstreamHandler {
             .addListener(ChannelFutureListener.CLOSE);
     }
 
+
+    private static Pattern PATTERN = Pattern.compile("^https?://");
     private static boolean isSameOrigin(final String origin, final String host) {
         if (Strings.isNullOrEmpty(host) == false) {
             // strip protocol from origin
-            final String originDomain = origin.replaceFirst("(http|https)://", "");
+            final String originDomain = PATTERN.matcher(origin).replaceFirst("");
             if (host.equals(originDomain)) {
                 return true;
             }
```

The same benchmark showed this to be around 250% faster and creates one less object.